lib/connection.js

```diff
@@ -328,15 +362,22 @@ class Connection extends EventEmitter {
     this.emit('_timeout');
   }
 
+  // Prepare a JSTP message to be sent over this connection
+  //   message - a message to prepare
+  //
+  _prepareMessage(message) {
+    return serde.stringify(message);
+  }
+
   // Send a JSTP message over this connection
-  //   message - a message to send
+  //   preparedMessage - a prepared message to send
+  //   message - a message object (used for development events thus optional)
   //
-  _send(message) {
-    const data = serde.stringify(message);
-    this.transport.send(data);
+  _send(preparedMessage, message) {
```

Comment:
Yes, it will avoid passing both parsed and not parsed arguments to the `_send` method by parsing the stringified message every time, which, as I've already said, makes no sense to me.